java/src/test/java/com/google/oak/verification/MainVerifierTest.java

```diff
@@ -0,0 +1,122 @@
+import com.google.oak.attestation.v1.LayerEvidence;
+import com.google.oak.attestation.v1.LayerReferenceValues;
+import com.google.oak.attestation.v1.ReferenceValues;
+import com.google.oak.attestation.v1.RootLayerEndorsements;
+import com.google.oak.attestation.v1.RootLayerEvidence;
+import com.google.oak.attestation.v1.RootLayerReferenceValues;
+import com.google.oak.attestation.v1.TeePlatform;
+import com.google.oak.attestation.v1.TransparentReleaseEndorsement;
+import com.google.oak.attestation.v1.VerifyLogEntry;
+import com.google.protobuf.ByteString;
+import java.lang.ref.Reference;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class MainVerifierTest {
+  private static final String LOG_ENTRY_PATH =
+      "oak_remote_attestation_verification/testdata/logentry.json";
+  private static final String ENDORSER_PUBLIC_KEY_PATH =
+      "oak_remote_attestation_verification/testdata/oak-development.pem";
+  private static final String REKOR_PUBLIC_KEY_PATH =
+      "oak_remote_attestation_verification/testdata/rekor_public_key.pem";
+  private static final String ENDORSEMENT_PATH =
+      "oak_remote_attestation_verification/testdata/endorsement.json";
+
+  private byte[] logEntryBytes;
+  private byte[] endorserPublicKeyBytes;
+  private byte[] rekorPublicKeyBytes;
+  private byte[] endorsementBytes;
+
+  @Before
+  public void setUp() throws Exception {
+    logEntryBytes = Files.readAllBytes(Path.of(LOG_ENTRY_PATH));
+    endorserPublicKeyBytes =
+        SignatureVerifier.convertPemToRaw(Files.readString(Path.of(ENDORSER_PUBLIC_KEY_PATH)));
+    rekorPublicKeyBytes =
+        SignatureVerifier.convertPemToRaw(Files.readString(Path.of(REKOR_PUBLIC_KEY_PATH)));
+    endorsementBytes = Files.readAllBytes(Path.of(ENDORSEMENT_PATH));
+  }
+
+  private TransparentReleaseEndorsement createTREndorsement() {
+    ByteString endorsement = ByteString.copyFrom(endorsementBytes);
+    ByteString logEntry = ByteString.copyFrom(logEntryBytes);
+    return TransparentReleaseEndorsement.newBuilder()
+        .setEndorsement(endorsement)
+        .setRekorLogEntry(logEntry)
+        .build();
+  }
+
+  private static Evidence createEvidence() {
+    return Evidence.newBuilder()
+        .setRootLayer(RootLayerEvidence.newBuilder()
+                          .setPlatform(TeePlatform.AMD_SEV_SNP)
+                          .setRemoteAttestationReport(ByteString.copyFromUtf8("whatever")))
+        .addLayers(
+            LayerEvidence.newBuilder().setEcaCertificate(ByteString.copyFromUtf8("whatever")))
+        .build();
+  }
+
+  private Endorsements createEndorsements() {
+    return Endorsements.newBuilder()
+        .setRootLayer(
+            RootLayerEndorsements.newBuilder().setStage0Endorsement(createTREndorsement()))
```

Comment:
I guess in real world the endorsement will be different for each layer?

Reply:
Yes!